src/ric/22380.c

```diff
@@ -33,7 +33,157 @@ void func_8015E7B4(Unkstruct_8010BF64* arg0) { // !FAKE:
 
 INCLUDE_ASM("asm/us/ric/nonmatchings/22380", func_8015E800);
 
-INCLUDE_ASM("asm/us/ric/nonmatchings/22380", func_8015EE28);
+void func_8015EE28(void) {
+    Collider sp10;
+    s32 temp_fp;
+    u32 temp_s0;
+    s32 temp_v1;
+    s32 var_a0;
+    s32 i;
+    u16 var_a1;
+    s16 temp_s16;
+
+    s16 newY;
+
+    s16 argX;
+    s16 argY;
+
+    u32 sp10effects;
+
+    u16* yPosPtr = &PLAYER.posY.i.hi;
+    u16* xPosPtr = &PLAYER.posX.i.hi;
+    s32* vram_ptr = &g_Player.pl_vram_flag;
+    // weird thing where i has to get initialized first
+    i = 1;
+
+    if (D_80097418 != 0) {
+        return;
+    }
+    for (i = 1; i < 4; i++) {
+        var_a0 = g_Player.colliders2[i].effects;
+        temp_s0 = var_a0 & 0x8801;
+        if ((var_a0 & 0x40)) {
+            continue;
+        }
+        if ((temp_s0 == 1) || (var_a0 & 0x8000)) {
+            if (((PLAYER.step == 0x1A) || (PLAYER.step == 0x17)) &&
+                !(var_a0 & 1)) {
+                continue;
+            }
+            argX = *xPosPtr + D_801545E4[i].unk0;
+            argY = *yPosPtr + D_801545E4[i].unk2 +
+                   g_Player.colliders2[i].unk10 + 1;
+            g_api.CheckCollision(argX, argY, &sp10, 0);
+            sp10effects = sp10.effects;
+            if (!(sp10effects & 1)) {
+                if ((g_Player.colliders2[i].effects != 1) ||
+                    (PLAYER.velocityY <= 0)) {
+                    *vram_ptr |= 2;
+                    if (!(*vram_ptr & 1) && ((g_Player.unk04 & 0x41) != 0x41)) {
+                        if (g_Player.colliders2[i].effects & 0x8000) {
```

Comment:
Yep will go through all of these, just wanted to first make sure it passed the functional CI tests, will update with a new commit in a bit.

As far as renaming EFFECT_ to COL_, that sounds reasonable, but should be its own PR since it will affect so much of the repo.